File: packages/gbnf-experiment/data/20260909T124157Z_4ee67ba2/ported_implementation/tests/unit/rules-builder/parse-name.test.ts
import assert from 'node:assert/strict';
import { describe, test } from 'node:test';
import { PARSE_NAME_ERROR, parseName } from '../../../src/rules-builder/parse-name.ts';
import { GrammarParseError } from '../../../src/utils/errors/index.ts';

describe('parse name', () => {
  test('should return correct name when encountering a valid name', () => {
    const src = 'validName';
    assert.equal(parseName(src, 0), src);
  });

  test('should return correct name when encountering a valid name starting at a non zero position', () => {
    assert.equal(parseName('123validName', 3), 'validName');
  });

  test('should throw error when encountering an invalid name', () => {
    const src = '123';
    assert.throws(
      () => parseName(src, 0),
      (err: unknown) => {
        assert.ok(err instanceof GrammarParseError);
        assert.equal(String(err), String(new GrammarParseError(src, 0, PARSE_NAME_ERROR)));
        return true;
      },
    );
  });
});

File: proposed-projects/round-trip-experiment/reverse/20260909T002206Z_273cf6b9/20260910T042804Z_e72f6c04/ported_implementation/gbnf/rules_builder/parse_name_test.py
import pytest

from ..utils.errors import GrammarParseError
from .parse_name import PARSE_NAME_ERROR, parse_name


def test_should_return_correct_name_when_encountering_a_valid_name():
    src = "validName"
    assert parse_name(src, 0) == src


def test_should_return_correct_name_when_starting_at_a_non_zero_position():
    src = "123validName"
    assert parse_name(src, 3) == "validName"


def test_should_raise_error_when_encountering_an_invalid_name():
    src = "123"
    with pytest.raises(GrammarParseError) as e:
        parse_name(src, 0)
    assert str(e.value) == str(GrammarParseError(src, 0, PARSE_NAME_ERROR))
